src/deploy/extensions/tasks.ts

```diff
@@ -0,0 +1,103 @@
+import * as clc from "cli-color";
+
+import * as extensionsApi from "../../extensions/extensionsApi";
+import * as refs from "../../extensions/refs";
+import * as utils from "../../utils";
+import { ErrorHandler } from "./errors";
+import { OperationType } from "../functions/tasks";
+import { InstanceSpec } from "./planner";
+
+export type DeploymentType = "create" | "update" | "delete";
+export interface ExtensionDeploymentTask {
+  run: () => Promise<void>;
+  spec: InstanceSpec;
+  type: DeploymentType;
+}
+export function extensionsDeploymentHandler(
+  errorHandler: ErrorHandler
+): (task: ExtensionDeploymentTask) => Promise<any | undefined> {
+  return async (task: ExtensionDeploymentTask) => {
+    let result;
+    try {
+      result = await task.run();
+    } catch (err) {
+      if (err.status == 429) {
```

Comment:
Wondering why 429 specifically is being handled? Is there some reason this is likely to happen? May be worth documenting.

Reply:
Refactored this a bit to be more clear. 429 (and, as I'm realizing now, 409) errors are retryable errors, so we should retry then with backoff. This handler function is used with queue.ts, which retries when the handler throws an error.

For all other error codes, we don't want to retry, so we instead record and return.

Reply:
In a larger sense - I'm a bit concerned about hitting Cloud Build quotas when deploying a lot of extensions at once. I'm planning on making some backend changes to make sure that we pipe through 429's that we get from DM, so that we would retry them here